### PR TITLE
Change _check_intervals from PhaseBackgroundMaker

### PR DESCRIPTION
**Pull request by MRegeard**
**Description**

This pull request slightly modify _check_interval from PhaseBackgroundMaker to satisfy more phase definition, namely being able to define phase outside of the [0,1], such as [0.8, 1.2] or [-0.2, 1.2] to define a phase range from [[0.8, 1], [0,0.2].

**Dear reviewer**

### CODE REVIEW

File: gammapy/makers/background/phase.py [hunk 1]
```diff
@@ -120,13 +120,13 @@ def run(self, dataset, observation):
 
     @staticmethod
     def _check_intervals(intervals):
-        """Split phase intervals that go beyond phase 1"""
+
```
Comment (bkhelifi) [inline gammapy/makers/background/phase.py:123]: A docstring is still needed...

**Review by bkhelifi — COMMENTED** [review]
Thanks @MRegeard!
I have just a comment about the docstring to write

File: gammapy/makers/background/phase.py [hunk 2]
```diff
@@ -108,25 +105,25 @@ def run(self, dataset, observation):
 
         acceptance_off = RegionNDMap.from_geom(geom=dataset.counts.geom)
         acceptance_off.data = np.sum([_[1] - _[0] for _ in self.off_phase])
-
+
```
Comment (QRemy) [inline gammapy/makers/background/phase.py:108]: Some formatting is needed. 
You can run black on each of the modified scripts or install a pre-commit hook as explained here : 

https://docs.gammapy.org/1.0/development/intro.html#:~:text=We%20adhere%20to,pre%2Dcommit%20install

**Review by QRemy — CHANGES_REQUESTED** [review]
Thanks @MRegeard, no comments on the content but some formatting is needed as explained in my comments.

File: gammapy/makers/background/tests/test_phase.py [hunk 3]
```diff
@@ -61,7 +61,11 @@ def test_run(observations, phase_bkg_maker):
...
         {"p_in": [[0.2, 0.3]], "p_out": [[0.2, 0.3]]},
         {"p_in": [[0.9, 0.1]], "p_out": [[0.9, 1], [0, 0.1]]},
+        {"p_in": [[-0.2, 0.1]], "p_out": [[0.8, 1], [0, 0.1]]},
+        {"p_in": [[0.8, 1.2]], "p_out": [[0.8, 1], [0, 0.2]]},
+        {"p_in": [[0.2, 0.4], [0.8, 0.9]], "p_out":[[0.2, 0.4], [0.8, 0.9]]}, 
     ],
 )
 def test_check_phase_intervals(pars):
-    assert PhaseBackgroundMaker._check_intervals(pars["p_in"]) == pars["p_out"]
+    #assert PhaseBackgroundMaker._check_intervals(pars["p_in"]) == pars["p_out"]
```
[the remark this answers is not in the document]
Comment (MRegeard) [reply gammapy/makers/background/tests/test_phase.py:70]: Sure, I wasn't sure that my test was working so I leave the former test as a comment, but I will remove it !

File: gammapy/makers/background/phase.py (hunk 2 above)
Comment (MRegeard) [reply gammapy/makers/background/phase.py:108]: I am not sure to understand what you mean here

Comment (QRemy) [reply gammapy/makers/background/phase.py]: The whitespace have to be removed to follow the codestyle convention.
To have this done automatically on before commit install a pre-commit hook in you gammapy folder :
`pip install pre-commit`
`pre-commit install`
here the commit is already done so I think that you need to apply the formatter directly
`pip install black`
`black gammapy/makers/background/phase.py`

Comment (MRegeard) [reply gammapy/makers/background/phase.py]: Okay I see thanks

Comment (QRemy) [reply gammapy/makers/background/phase.py]: I just did it but you know for next time ;)

**Review by QRemy — APPROVED** [review]

**Pull request merged**